spec/javascripts/selectorSpec.js

```diff
@@ -37,4 +37,45 @@ describe("Suite for selector", function() {
     updateSelector("a", []);
     expect(d3.select("#aSelector").property("value")).toEqual("");
   });
+
+  it("should update clusterSelector when filterSelector==Cluster", function() {
+    d3.select("body")
+        .append("select")
+        .attr("id", "clusterSelector");
+    d3.select("body")
+        .append("select")
+        .attr("id", "filterSelector");
+
+    updateSelector("cluster", ["All", "cat", "dog"]);
+    updateSelector("filter", ["Cluster", "Zone"]);
+    d3.select("select#clusterSelector").property("value", "cat");
+
+    clusters = ["All", 0, 1];
```

Comment:
I don't think "clusters" is defined in this test. It looks like it's initialized in a different test, but we shouldn't assume that tests are run in any particular order. I recommend defining clusters as a mutable variable "let clusters = ..." at the beginning of the describe call. That way, you can use the variable in each of the tests.

Reply:
The reason I defined clusters and allZones (in a different test case) like that is because I wanted them to be global variables since those are the parameters that updateFilter() gives to updateSelector().

Reply:
Is there a better to do this?